### PR TITLE
Ofir petrushka/cookbook name depends

### PR DESCRIPTION
**Pull request by ofir-petrushka**
Use cookbook_name over hardcoding the cookbook name 'newrelic-ng'

### CODE REVIEW

File: providers/nrsysmond.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
   r = template 'nrsysmond.cfg' do
     path      new_resource.config_file
-    cookbook  new_resource.cookbook
+    cookbook  cookbook_name
```
Comment (chr4) [inline providers/nrsysmond.rb:32]: Do you know when `cookbook_name` was introduced? Will this break older `chef-clients` that are in use still?
Does this variable always resolve to `newrelic-ng`, even when the provider is called from somewhere else?

File: recipes/default.rb [hunk 2]
```diff
@@ -18,4 +18,4 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-include_recipe 'newrelic-ng::nrsysmond-default'
+include_recipe cookbook_name + '::nrsysmond-default'
```
Comment (chr4) [inline recipes/default.rb:21]: Do you think it's in general a wise idea to never hardcode the cookbooks name anywhere?
Can you explain why this is advantageous - I've not seen this practise yet in other cookbooks.

File: resources/php_agent.rb [hunk 3]
```diff
@@ -28,7 +28,6 @@
 
 attribute :mode,           kind_of: [Integer, String], default: 0o644
 
-attribute :cookbook,       kind_of: String, default: 'newrelic-ng'
```
Comment (chr4) [inline resources/php_agent.rb:31]: This removes the ability of providing a custom source from the caller's cookbook. I'm not sure if this is used by anyone, but at least in theory this provides a bit more flexibility. I'm not using `php_agent` myself, so if you say that it makes no sense to have a custom `newrelic.ini.php`, then I'm willing to drop this.

File: recipes/default.rb (hunk 2 above)
Comment (jeffbyrnes) [inline recipes/default.rb:21]: I’d also argue against this. There’s really no reason not to explicitly call the recipe.

File: resources/php_agent.rb (hunk 3 above)
Comment (jeffbyrnes) [inline resources/php_agent.rb:31]: I’d argue the other way: there are plenty of reasons an operator might supply their own `newrelic.ini.php`; I’ve used this feature myself when we still used this cookbook for our New Relic integration.

**Comment by ofir-petrushka** [comment]
It's 2 against one so feel free to close it.
As well as it's not really important to me.

I'd explain my point anyway if instreasted in reading.

For normal programming languages if you find:
file a: line x: echo 'hello world'
file b: line y: echo 'hello world'
file c: line z: echo 'hello world'

you would think of putting some common
message='hello world'
and update 
file a: line x: echo message
file b: line y: echo message
file c: line z: echo message
or even a function for this example... 
Or maybe you rather of 'this.' in a class as this is more like a function call.

I have been made aware that it's also possible to change 
include_recipe 'newrelic-ng::nrsysmond-default'
to 
include_recipe '::nrsysmond-default'

I don't know where to find which chef version things were added in. At least not in an easy fast manner. (I know it's in the repo ect.)

I don't know well the resources/providers to comment on that.
I only know what is in the current tests pass at least for external-php-ubuntu-12.04.

I thought this might be useful for others cause I needed it, but as I said feel free to close it.

**Comment by chr4** [comment]
Ok, closing this then. Thanks for the request, nonetheless!

**Pull request closed without merging**